core/camel-support/src/main/java/org/apache/camel/support/component/AbstractApiProducer.java

```diff
@@ -113,12 +113,14 @@ public void interceptProperties(Map<String, Object> properties) {
     /**
      * Invoke the API method. Derived classes can override, but MUST call super.doInvokeMethod().
      *
+     * @param  exchange              the current exchange
      * @param  method                API method to invoke.
      * @param  properties            method arguments from endpoint properties and exchange In headers.
      * @return                       API method invocation result.
      * @throws RuntimeCamelException on error. Exceptions thrown by API method are wrapped.
      */
-    protected Object doInvokeMethod(ApiMethod method, Map<String, Object> properties) throws RuntimeCamelException {
+    protected Object doInvokeMethod(Exchange exchange, ApiMethod method, Map<String, Object> properties)
+            throws RuntimeCamelException {
         return ApiMethodHelper.invokeMethod(endpoint.getApiProxy(method, properties), method, properties);
     }
```

Comment:
what about keeping a doInvokeMethod(ApiMethod method, Map<String, Object> properties) potentially marked as deprecated to avoid the API breakage?